### PR TITLE
Implementation to save and resume current game state <feature, user-interface>

### PR DESCRIPTION
**Pull request by asgill**
The current game state can be saved by pressing ‘Save’ button, and can
be resumed lated using ‘Resume’ button.

### CODE REVIEW

File: src/main/java/nl/tudelft/bejeweled/game/BejeweledGame.java [hunk 1]
```diff
@@ -32,6 +46,10 @@
     private BoardFactory boardFactory;
 
     private int score = 0;
+    
+    private int isStop = 0;
```
Comment (lange88) [inline src/main/java/nl/tudelft/bejeweled/game/BejeweledGame.java]: Why make these int variables? Isn't it more clear to use a boolean?

**Comment by lange88** [comment]
The code works fine as far as I can tell but maybe we should make the game save the state when the user exits the game either by clicking the cross or the exit button (and a game is in progress) and then ask the user if he wants to load the previous state when he starts the laucher next time?

Other than that nice work!

**Comment by asgill** [comment]
It saves the game when you click exit button. I am trying to figure out how to save when x button is clicked. Also resume only works when you start the game or after you press the save button, though I am trying to display some text as well in the start of the game

**Comment by lange88** [comment]
Check out this link if you're still trying to fix it:
http://www.java2s.com/Code/Java/JavaFX/Stagecloseevent.htm

```
stage.setOnCloseRequest(new EventHandler<WindowEvent>() {
          public void handle(WindowEvent we) {
              System.out.println("Stage is closing");
          }
      });    
```
I'm not 100% sure in what situations this handler is called, for example I can imagine clicking the "exit" button could also trigger the event. In that case you only need to put the code for saving the state in the close request handler and not in the exit button click handler..

**Comment by asgill** [comment]
It is done. Thanks! Link was helpful

**Pull request merged**